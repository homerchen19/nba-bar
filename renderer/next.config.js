const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
