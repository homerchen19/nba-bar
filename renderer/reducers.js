import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
