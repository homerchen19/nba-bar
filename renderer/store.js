import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
