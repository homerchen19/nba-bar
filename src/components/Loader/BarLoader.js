import React from 'react';
import styled, { keyframes } from 'styled-components';

import { colors } from '../../styles/theme';

const Wrapper = styled.div`
  position: relative;
  width: 100vw;
  height: 6px;
  background-color: ${colors.blue};
`;

const loading = keyframes`
  from {
    z-index: 100;
    left: 50%;
    width: 0;
  }

  50% {
    z-index: 10;
    left: 0;
    width: 100%;
  }

  to {
    left: 0;
    width: 100%;
  }
`;

const Bar = styled.div`
  content: '';
  display: inline;
  position: absolute;
  left: 50%;
  width: 0;
  height: 100%;
  text-align: center;
`;

const FirstBar = Bar.extend`
  background-color: ${colors.red};
  animation: ${loading} 2s ease-in-out infinite;
`;

const SecondBar = Bar.extend`
  background-color: ${colors.blue};
  animation: ${loading} 2s ease-in-out 1s infinite;
`;

const BarLoader = () => (
  <Wrapper>
    <FirstBar />
    <SecondBar />
  </Wrapper>
);

export default BarLoader;
