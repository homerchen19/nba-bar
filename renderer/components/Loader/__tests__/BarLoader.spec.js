import React from 'react';
import renderer from 'react-test-renderer';

import BarLoader from '../BarLoader';

it('be defined', () => {
  expect(BarLoader).toBeDefined();
});

it('match snapshot', () => {
  const tree = renderer.create(<BarLoader />).toJSON();

  expect(tree).toMatchSnapshot();
});
