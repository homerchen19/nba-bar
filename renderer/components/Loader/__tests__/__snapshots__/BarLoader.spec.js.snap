// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="BarLoader__Wrapper-sc-1cjmyxn-0 cthRyu"
>
  <div
    className="BarLoader__FirstBar-sc-1cjmyxn-2 ivdCQi"
  />
  <div
    className="BarLoader__SecondBar-sc-1cjmyxn-3 dktTAd"
  />
</div>
`;
